python-avd/pyavd/schema/metaschema/meta_schema_model.py

```diff
@@ -0,0 +1,559 @@
+
+    Also covers internal attributes and methods used for documentation generation.
+    All these are prefixed with underscore.
+    """
+
+    class ConvertType(str, Enum):
+        int = "int"
+        str = "str"
+
+    # AvdSchema field properties
+    type: Literal["bool"]
+    convert_types: List[ConvertType] | None = None
+    """List of types to auto-convert from. For 'bool' auto-conversion is supported from 'int' and 'str'"""
+    default: bool | None = None
+    """Default value"""
+    valid_values: List[bool] | None = None
+    """List of valid values"""
+    dynamic_valid_values: str | None = None
+    """
+    Path to variable under the parent dictionary containing valid values.
+    Variable path use dot-notation and variable path must be relative to the parent dictionary.
+    If an element of the variable path is a list, every list item will be unpacked.
+    Note that this is building the schema from values in the _data_ being validated!
+    """
+
+    # Type of schema docs generators to use for this schema field.
+    _table_row_generator = TableRowGenBool
+    _yaml_line_generator = YamlLineGenBool
+
+
+class AvdSchemaStr(AvdSchemaBaseModel):
+    """
+    Pydantic model for AvdSchema fields of type "str".
+
+    Contains fields that applies to this type specifically. Other fields are inherited from the base class.
+
+    Also covers internal attributes and methods used for documentation generation.
+    All these are prefixed with underscore.
+    """
+
+    class ConvertType(str, Enum):
+        bool = "bool"
+        int = "int"
+        float = "float"
+
+    class Format(str, Enum):
+        ipv4 = "ipv4"
+        ipv4_cidr = "ipv4_cidr"
+        ipv6 = "ipv6"
+        ipv6_cidr = "ipv6_cidr"
+        ip = "ip"
+        cidr = "cidr"
+        mac = "mac"
+
+        def __str__(self):
+            return self.value
+
+    # AvdSchema field properties
+    type: Literal["str"]
+    convert_to_lower_case: bool | None = False
+    """Convert string value to lower case before performing validation"""
+    convert_types: List[ConvertType] | None = None
+    """List of types to auto-convert from.\n\nFor 'str' auto-conversion is supported from 'bool' and 'int'"""
+    default: str | None = None
+    """Default value"""
+    format: Format | None = None
+    """String format"""
+    min_length: int | None = None
+    """Mininmum string length"""
+    max_length: int | None = None
+    """Maximum string length"""
+    pattern: str | None = None
+    """
+    A regular expression which will be matched on the variable value.
+    The regular expression should be valid according to the ECMA 262 dialect.
+    Remember to use double escapes.
+    """
+    valid_values: List[str] | None = None
+    """List of valid values"""
+    dynamic_valid_values: str | None = None
+    """
+    Path to variable under the parent dictionary containing valid values.
```

Comment:
Can we remove the limitation of parent dict with this - by referencing full path in schema from the store? (maybe a later feature ;) )